lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -545,84 +546,203 @@ def as_dict_with_promql_labels(self):
 
     def render(self, template: str):
         """Render a juju-topology template string with topology info."""
-        return template.replace("%%juju_topology%%", self.promql_labels)
+        # TODO remove, once https://github.com/canonical/prometheus-operator/pull/166 is merged
+        return template.replace(JujuTopology.STUB, self.promql_labels)
+
+
+class InvalidAlertRulePathError(Exception):
+    """Raised if the alert rules folder cannot be found or is otherwise invalid."""
 
+    def __init__(
+        self,
+        alert_rules_absolute_path: Path,
+        message: str,
+    ):
+        self.alert_rules_absolute_path = alert_rules_absolute_path
+        self.message = message
 
-def load_alert_rule_from_file(path: Path, topology: JujuTopology) -> Optional[dict]:
-    """Load alert rule from a rules file.
+        super().__init__(self.message)
+
+
+class AlertRules:
+    """Data store for prometheus alert rules.
```

Comment:
Looking at the methods defined in this class (even though they are static methods) the responsibility of this class seems to be
1. Read alert rules from a directory
2. Ensure alert rules have topology labels
Hence it I am not sure if it correct to say the responsibility of this class is just to store alert rules.